### PR TITLE
Fix/re-enable MSVC compiler warnings

### PR DESCRIPTION
**Pull request by Ansariel**
@marchcat Instead of some pointless whitespace changes 😆 here's something that actually might help code quality: Re-enabled a bunch of compiler warnings that do not occur anymore and also address the "possible loss of data" warnings by using the correct type or adding explicit casts.

### CODE REVIEW

**Comment by marchcat** [comment]
Thank you, @Ansariel!

Even a mere 295-files change looks too huge for Github to display the diff, so I'll try to merge it locally first.

File: indra/newview/llinventorymodel.cpp [hunk 1]
```diff
@@ -1290,7 +1290,6 @@ void LLInventoryModel::collectDescendentsIf(const LLUUID& id,
         }
     }
 
-    LLViewerInventoryItem* item = NULL;
```
Comment (marchcat) [inline indra/newview/llinventorymodel.cpp:1301]: Thanks! I didn't see this in the build log for some reason.

Comment (Ansariel) [reply indra/newview/llinventorymodel.cpp:1301]: I didn't break the Windows build though...

Comment (marchcat) [reply indra/newview/llinventorymodel.cpp:1301]: Right, it looks like GitHub is trying to build the non-opensource version on your external repo.
I'll test it on my dev setup later today.

**Comment by marchcat** [comment]
Since I can't make a PR to your repo for some reason, please cherry-pick [this commit](https://github.com/secondlife/viewer/commit/0d744b39a6eeda5e1e1c8513f2c581037cff79c7) from the branch `marchcat/buildfix-1613`. It should fix the build.

**Comment by Ansariel** [comment]
> Since I can't make a PR to your repo for some reason, please cherry-pick [this commit](https://github.com/secondlife/viewer/commit/0d744b39a6eeda5e1e1c8513f2c581037cff79c7) from the branch `marchcat/buildfix-1613`. It should fix the build.

OK done - I removed the obsolete 32bit Windows package though. :)

**Comment by Ansariel** [comment]
Does the update source package also contain the changes from BOOL to bool in llpathinglib.h?

**Comment by marchcat** [comment]
> Does the update source package also contain the changes from BOOL to bool in llpathinglib.h?

Yes, they were included in the previous build.

**Comment by Ansariel** [comment]
> > Does the update source package also contain the changes from BOOL to bool in llpathinglib.h?
> 
> Yes, they were included in the previous build.

When can we get updated stub and TPV versions then, so we can remove the take BOOL typedefs?

**Comment by marchcat** [comment]
> When can we get updated stub and TPV versions then, so we can remove the take BOOL typedefs?

As far as I'm aware, there still are some issues related to those builds..

**Comment by Ansariel** [comment]
So, apparently the 3p for macOS isn't available now.

**Comment by marchcat** [comment]
Probably, the `llphysicsextensions_source` package for windows isn't available for this repo as well, mac build just fails first and cancels the rest.
However, I tried to build it and it finished fine.

**Comment by Ansariel** [comment]
So you can merge it and fix any issue afterwards if any arise.

**Comment by marchcat** [comment]
Right. I'd just like to do a couple more test merges/builds first.

**Review by marchcat — APPROVED** [review]

File: indra/llcommon/llpreprocessor.h [hunk 2]
```diff
@@ -122,24 +122,6 @@
...
-#pragma warning (disable : 4180)    // qualifier applied to function type has no meaning; ignored
-//#pragma warning( disable : 4284 ) // silly MS warning deep inside their <map> include file
-
-#if ADDRESS_SIZE == 64
-// That one is all over the place for x64 builds.
-#pragma warning( disable : 4267 )   // 'var' : conversion from 'size_t' to 'type', possible loss of data)
-#endif
-
-#pragma warning( disable : 4503 )   // 'decorated name length exceeded, name was truncated'. Does not seem to affect compilation.
-#pragma warning( disable : 4800 )   // 'BOOL' : forcing value to bool 'true' or 'false' (performance warning)
```
Comment (brad-linden) [inline indra/llcommon/llpreprocessor.h:142]: YESS!  thank you!

**Review by brad-linden — APPROVED** [review]

**Comment by Ansariel** [comment]
> Right. I'd just like to do a couple more test merges/builds first.

How did the testing go? :)

**Comment by marchcat** [comment]
> How did the testing go? :)

The change works fine on my side, despite failing the build here.
However, I may need additional time for integration due to frequent blackouts..

**Comment by Ansariel** [comment]
> > How did the testing go? :)
> 
> The change works fine on my side, despite failing the build here. However, I may need additional time for integration due to frequent blackouts..

Would be good if this could get in before materials-featurette gets merged.

**Comment by marchcat** [comment]
Right, that's the plan.

**Pull request merged**